src/repository.h

```diff
@@ -142,6 +143,9 @@ struct git_repository {
 
 	git_atomic attr_session_key;
 
+	git_hook_execution_cb hook_executor;
+	void *hook_payload;
```

Comment:
Hm. There's some memory management issues with the payload here. When the repository is free'd, we'll loose the payload if the caller does not explicitly free it for himself, which is rather cumbersome. We'd need to have either some function `hook_free` which can be registered with the repository or another API approach here. Will discuss in a later comment.